src/bokeh/core/property/datetime.py

```diff
@@ -119,6 +120,11 @@ def validate(self, value: Any, detail: bool = True) -> None:
     def is_timestamp(value: Any) -> bool:
         return isinstance(value, (float,) + bokeh_integer_types) and not isinstance(value, bool)
 
+class FloatOrDatetime(Datetime):
```

Comment:
I figured out that the `Datetime` type was allready accepting `float` and `datetime` objects. So this type was perfect. But the name was ambiguous, so I decided to make an alias named `FloatOrDatetime`, as @mattpap suggested. I Hope I got the idea correct. If `FloatOrDatetime` is to long or if any other name is better, I am happy for any ideas.

Reply:
I think I'd be happier with this if `Datetime` could be restricted and then `FloatOrDatetime = Either(Float, Datetime)` but that might be too disruptive. What are your thoughts, @mattpap ?

Reply:
> decided to make an alias named `FloatOrDatetime`, as @mattpap suggested

By alias I actually meant having something like this in a common location:
```py
CoordinateLike = Either(Float, DateTime, Factor)
```
and used like this:
```py
class Label(Annotation):
    x = CoordinateLike(help="x-coordinate")
    y = CoordinateLike(help="y-coordinate")
```
or
```py
class BoxAnnotation(Annotation):

    left = Nullable(CoordinateLike, help="left")
    right = Nullable(CoordinateLike, help="right")
```
Since 3.0 we can create aliases by just creating an instance of a property type like `Either(...)`. This then can be used to construct a specific property by calling the alias or using it as part of other property types.

Reply:
@mattpap At the moment `NumberSpec` isn't touched. Do you think should be adapted, too? I was afraid that the changes are going to be to large.